### PR TITLE
RTOS rewrite

### PR DESCRIPTION
**Pull request by kylestach**
Rewrite the entire code base to use RTOS.

Design document: https://docs.google.com/document/d/1767ShGdEDZaqAaYE4M8tanvGCliVxE0DQHzV9dSy7es/edit#

I'm so sorry for the size of this PR.

### CODE REVIEW

File: robot/control/Src/modules/RadioModule.cpp [hunk 1]
```diff
@@ -17,45 +18,69 @@ RadioModule::RadioModule(BatteryVoltage *const batteryVoltage,
...
+    {
+        auto batteryVoltageLock = batteryVoltage.lock();
+        auto fpgaStatusLock = fpgaStatus.lock();
+        auto robotIDLock = robotID.lock();
+        auto kickerInfoLock = kickerInfo.lock();
+
+        // Just check to see if our robot id is valid
+        // That way we don't conflict with other robots on the network
+        // that are working
+//        if (batteryVoltageLock->isValid && fpgaStatusLock->isValid && robotIDLock->isValid) {
```
Comment (JNeiger) [inline robot/control/Src/modules/RadioModule.cpp]: Depending on if the robotID is read before this or not, it's possible that this robot will connect to soccer as `robotID=0` for a single frame then swaps to the correct robotID. I believe soccer starts to act weird as more MAC<->RobotID combinations connect to soccer. The robots start to be only intermittently connected and you have to restart soccer to fix

File: robot/control/Inc/iodefs.h [hunk 2]
```diff
@@ -12,14 +12,16 @@
 #define RADIO_R1_INT  p18
 #define RADIO_GLB_RST p19
 
-#define FPGA_KICKER_SPI_BUS SpiBus::SpiBus3
+#define FPGA_SPI_BUS SpiBus::SpiBus3
```
Comment (CollinAvidano) [inline robot/control/Inc/iodefs.h]: Thank you so much

**Comment by kylestach** [comment]
This is _almost_ ready to be merged. I still need to verify that it doesn't break kicker functionality.

File: robot/control/Src/modules/RadioModule.cpp (hunk 1 above)
Comment (kylestach) [reply robot/control/Src/modules/RadioModule.cpp]: Fixed, this was only temporarily disabled.

**Pull request merged**